lte/gateway/c/sctpd/src/util.cpp

```diff
@@ -111,20 +111,22 @@ int set_sctp_opts(const int sd, const uint16_t instreams,
   return 0;
 }
 
-int convert_addrs(const InitReq* req, struct sockaddr** addrs, int* num_addrs) {
+int convert_addrs(const InitReq* req, struct sockaddr** ret_addrs,
+                  int* num_addrs) {
   int i;
+  struct sockaddr* addrs;
   struct sockaddr_in* ipv4_addr;
   struct sockaddr_in6* ipv6_addr;
 
   auto num_ipv4_addrs = req->ipv4_addrs_size();
   auto num_ipv6_addrs = req->ipv6_addrs_size();
   *num_addrs = num_ipv4_addrs + num_ipv6_addrs;
 
-  *addrs = (struct sockaddr*)calloc(*num_addrs, sizeof(struct sockaddr*));
-  if (*addrs == NULL) return -1;
+  addrs = (struct sockaddr*)calloc(*num_addrs, sizeof(struct sockaddr));
```

Comment:
for `sockadd_in6` this is still not large enough.

Reply:
good find.